BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -15,20 +15,43 @@ public struct ElementContent: Measurable {
         self.storage = builder
     }
 
-    public func measure(in constraint: SizeConstraint) -> CGSize {
-        return storage.measure(in: constraint)
+    public func measure(in constraint: SizeConstraint, environment: Environment) -> CGSize {
+        return storage.measure(in: constraint, environment: environment)
     }
 
     public var childCount: Int {
         return storage.childCount
     }
 
-    func performLayout(attributes: LayoutAttributes) -> [(identifier: ElementIdentifier, node: LayoutResultNode)] {
-        return storage.performLayout(attributes: attributes)
+    func performLayout(attributes: LayoutAttributes, environment: Environment) -> [(identifier: ElementIdentifier, node: LayoutResultNode)] {
+        return storage.performLayout(attributes: attributes, environment: environment)
     }
 
 }
 
+extension ElementContent {
+    func measurable(in environment: Environment) -> Measurable {
+        struct EnvironmentMeasurable: Measurable {
```

Comment:
Is the idea here that we have some measurables which do not conform to or have an environment? Assume this is to avoid having to do downstream refactoring?

Reply:
Yep, since ElementContent cannot be measured without an environment now, this is an adapter for functions that still want a Measurable.

I had an alternative PR where I tried refactoring Measurable instead, and the result was (IMO) worse.